share/getters/store.go

```diff
@@ -123,10 +123,11 @@ func (sg *StoreGetter) GetSharesByNamespace(
 	shares, err = collectSharesByNamespace(ctx, blockGetter, root, nID)
 	if errors.Is(err, ipld.ErrNodeNotFound) {
 		// convert error to satisfy getter interface contract
-		err = share.ErrNotFound
+		err = share.ErrNamespaceNotFound
```

Comment:
Q: It's possible to infer from just DAH whether there is namespaces data or not. Why aren't we doing this on client side?

Reply:
This is a great idea, but not complete: The client side check can skip network requests if the nID isn't even found in the range of nIDs that the root covers. However, it can still be the case that an nID exists in the root that does not actually have any data (e.g. if a root contains the nids nID-1...nid+1)

I propose the following diff:
```diff
index c93a138e..805dae20 100644
--- a/share/getters/shrex.go
+++ b/share/getters/shrex.go
@@ -122,6 +122,13 @@ func (sg *ShrexGetter) GetSharesByNamespace(
                attempt int
                err     error
        )
+
+       // verify that the namespace could exist inside the roots before starting network requests
+       roots := filterRootsByNamespace(root, id)
+       if len(roots) == 0 {
+               return nil, share.ErrNamespaceNotFound
+       }
+
        for {
                if ctx.Err() != nil {
                        return nil, ctx.Err()
```

I don't want to include it in shrexnd client code, because the client should not care about such things, and also the logic already exists in the other getters as a quick-fail (through collectSharesByNamespace)

Reply:
Ok. Makse sense!